cpp/include/cuvs/cluster/agglomerative.hpp

```diff
@@ -163,15 +166,27 @@ struct mutual_reachability_params {
  */
 void build_linkage(
   raft::resources const& handle,
-  raft::device_matrix_view<const float, int, raft::row_major> X,
+  raft::device_matrix_view<const float, int64_t, raft::row_major> X,
+  std::variant<linkage_graph_params::distance_params,
+               linkage_graph_params::mutual_reachability_params> linkage_graph_params,
```

Comment:
We should probably add some docs for this new option and highlight how it differs from the distance_params.

Reply:
I'm kind of confused here because the `build_linkage` function already had two options (mutual reach and distance) for the params. We just have two different APIs now (one for host data and one for device data) 👀 The real new thing here is adding `cuvs::neighbors::all_neighbors::all_neighbors_params` to `mutual_reachability_params`